clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1554,6 +1575,13 @@ private ConfigEntry configEntry(CreatableTopicConfigs config) {
 
             @Override
             void handleFailure(Throwable throwable) {
+                // If there were any topics retries due to a quota exceeded exception, we propagate
+                // the initial error back to the caller if the request timed out.
+                if (options.shouldRetryOnQuotaViolation() && throwable instanceof TimeoutException) {
+                    completeQuotaExceededException(futures, quotaExceededExceptions,
```

Comment:
Perhaps we could make this `maybeCompleteQuotaExceededException` and pass in the throwable, so that we can do the check for TimeoutException in that helper method rather than in every `handleFailure`?

Reply:
That's a very good suggestion, thanks.